src/bp/core/services/stats-service.ts

```diff
@@ -34,53 +45,77 @@ export class StatsService {
     @inject(TYPES.CMSService) private cmsService: CMSService,
     @inject(TYPES.AuthService) private authService: AuthService,
     @inject(TYPES.UserRepository) private userRepository: UserRepository,
+    @inject(TYPES.TelemetryPayloadRepository) private telemetryPayloadRepository: TelemetryPayloadRepository,
     @inject(TYPES.Database) private database: Database
   ) {}
 
   public start() {
     // tslint:disable-next-line: no-floating-promises
-    this.run()
-    setInterval(this.run.bind(this), ms(JOB_INTERVAL))
+    this.run(this.getStats.bind(this), LOCK_RESOURCE, JOB_INTERVAL, `${telemetry1}`)
+    // tslint:disable-next-line: no-floating-promises
+    this.run(this.getBuiltinActionsStats.bind(this), LOCK_RESOURCE24, TELEMETRY_INTERVAL, `${telemetry2}`)
+
+    setInterval(
+      this.run.bind(this, this.getStats.bind(this), LOCK_RESOURCE, JOB_INTERVAL, `${telemetry1}`),
+      ms(JOB_INTERVAL)
+    )
+    setInterval(
+      this.run.bind(this, this.getBuiltinActionsStats.bind(this), LOCK_RESOURCE24, TELEMETRY_INTERVAL, `${telemetry2}`),
+      ms(TELEMETRY_INTERVAL)
+    )
+
+    // tslint:disable-next-line: no-floating-promises
+    this.refreshDB(DB_REFRESH_INTERVAL)
+
+    setInterval(this.refreshDB.bind(this, DB_REFRESH_INTERVAL), ms(DB_REFRESH_INTERVAL))
+  }
```

Comment:
I'm not sure I understand why it would be better to pass a string and then have a switch/case on it? I was trying to expand in the future

Reply:
That's something we call over-engineering :P Right now there's two jobs. If we add a lot more then we'll refactor it to handle it better. Always keep this in your mind: KIS. Keep It Simple. 

It's much easier to make a small refactor in the future when needed than trying to figure out what will be the possible expansions in the future and trying to plan it all in advance.

Reply:
@allardy I don't see where is the over engineering in there ... What you suggest is the exact same thing just a little less clean ...  so if you think that the current implementation is over-engineered then what you're suggesting is as over-engineered

A good suggestion would be to remove the "run" func and call 2 different job funcs directly ... this would remove the "over-engineering"